dbt/adapters/redshift/impl.py

```diff
@@ -114,6 +115,44 @@ def valid_incremental_strategies(self):
     def timestamp_add_sql(self, add_to: str, number: int = 1, interval: str = "hour") -> str:
         return f"{add_to} + interval '{number} {interval}'"
 
+    def _get_cursor(self):
```

Comment:
Yes, you are correct. The _get_cursor is used to call _get_tables. I've instantiated the cursor in  _get_cursor() instead of instantiating a new cursor every time we call the redshift_connector metadata methods. I'm working on overriding other macros with these metadata calls, which will also be using the same  _get_cursor(). That way I don't have to create a new connector.cursor in every one of these new methods.